### ISSUE TITLE
Some functions are not listed in "referenced by"

### ISSUE THREAD

**Issue by shosakam** [doxygen/doxygen#9783]
**Describe the bug**
Some functions are not listed in "referenced by".

**To Reproduce**
The following case, call_sub() is not listed in "referenced by" in sub() documentation.

```c
static void sub(void)
{
	// referenced by call_sub()
}

static void call_sub(SETTING_INFO p_info)
{
	// Commenting out either line avoids the issue.
	char *buf_ptr = (char *)p_info->p_param->p_ptr;
	int buf_distance_word[3] = {0};

	sub();
}

void main(void)
{
	call_sub(ginfo);
}
```
![screenshot](https://user-images.githubusercontent.com/101457785/211704689-86755961-8d19-45d7-bd29-7020c0a27eb3.jpg)

**Configuration**
EXTRACT_ALL            = YES
EXTRACT_STATIC         = YES
REFERENCED_BY_RELATION = YES

**Version**
doxygen 1.9.6
Windows11 22H2

**Comment by doxygen** [comment]
@albert-github I found a different way to fix this, which I expect to have less risk of unwanted side effects.

**Comment by albert-github** [comment]
An alternative code update has been made in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
